flake.nix

```diff
@@ -3,198 +3,131 @@
-              inherit url sha256;
-            };
-            doCheck = false;
-          };
-
-        lsprotocol = buildWheel {
-          pname = "lsprotocol";
-          version = "2024.0.0b1";
-          url = "https://files.pythonhosted.org/packages/4d/1b/526af91cd43eba22ac7d9dbdec729dd9d91c2ad335085a61dd42307a7b35/lsprotocol-2024.0.0b1-py3-none-any.whl";
-          sha256 = "sha256-k3hQUKwVWuK+FrHr+9dMIU/rPT73exA5nOlB5czvbr0=";
-          propagatedBuildInputs = with pkgs.python3Packages; [
-            attrs
-            cattrs
-            jsonschema
-          ];
-        };
-
-        bibtexparser2 = buildWheel {
-          pname = "bibtexparser";
-          version = "2.0.0b4";
-          url = "https://files.pythonhosted.org/packages/3f/64/6eda019a3b0d3aa6834865c126b66cfd442f56829befdca571c40662f8a6/bibtexparser-2.0.0b4-py3-none-any.whl";
-          sha256 = "sha256-FOWMfvltp4z2dce7LPFbA2E9Rj737c09ruIPU7SJzoM=";
-          propagatedBuildInputs = with pkgs.python3Packages; [ pylatexenc ];
-        };
-
-        # Custom Python packages
-        customPythonPackages = with pkgs.python3Packages; {
-          pyzotero = buildWheel {
-            pname = "pyzotero";
-            version = "1.5.25";
-            url = "https://files.pythonhosted.org/packages/cb/3c/717f90930fbba6ec433a8b6eb9c8854089b7cd54118fb3dd5822d53bdfc7/pyzotero-1.5.25-py3-none-any.whl";
-            sha256 = "sha256-ZSkTDLfH53OWPZLbfnoYtpipesgVZ2YyC1Xr1MfpTtU=";
-            propagatedBuildInputs = [
-              bibtexparser2
-              feedparser
-              pytz
-              requests
-            ];
+    let
+      inherit (nixpkgs) lib;
+      forAllSystems = lib.genAttrs lib.systems.flakeExposed;
+
+      # Load workspace from current directory
+      workspace = uv2nix.lib.workspace.loadWorkspace { workspaceRoot = ./.; };
+
+      # Create package overlay from workspace
+      overlay = workspace.mkPyprojectOverlay {
+        sourcePreference = "wheel";
+      };
+
+      # Python sets grouped per system
+      pythonSets = forAllSystems (
+        system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
+
+          # Base Python package set from pyproject.nix
+          baseSet = pkgs.callPackage pyproject-nix.build.packages {
+            python = pkgs.python312;
           };
 
-          pygls = buildWheel {
-            pname = "pygls";
-            version = "2.0.0a2";
-            url = "https://files.pythonhosted.org/packages/f8/47/7d7b3911fbd27153ee38a1a15e3977c72733a41ee8d7f6ce6dca65843fe9/pygls-2.0.0a2-py3-none-any.whl";
-            sha256 = "sha256-sgI2kyFAk0OqZEDXMRHZ+gwi5YBGb/HHaWuDWLuR8kM=";
-            propagatedBuildInputs = [ lsprotocol ];
+          # Build fixups for problematic packages
+          # Those packages are using setuptools, so we need to pass setuptools
+          # as a build input
+          pyprojectOverrides = final: prev: {
+            sgmllib3k = prev.sgmllib3k.overrideAttrs (old: {
```

Comment:
Out of curiosity, what is this `sgmllib3k` package? Do we have a dependency on it?

Reply:
It is because of `pyzotero`, it depends on `feedparser`, thus `sgmllib3k`

Reply:
Good to know, thanks for the investigation.